manim/mobject/mobject.py

```diff
@@ -100,18 +101,18 @@ def __init__(
         color: ParsableManimColor | list[ParsableManimColor] = WHITE,
         name: str | None = None,
         dim: int = 3,
-        target=None,
+        target: Mobject | None = None,
         z_index: float = 0,
     ) -> None:
-        self.name = self.__class__.__name__ if name is None else name
-        self.dim = dim
-        self.target = target
-        self.z_index = z_index
+        self.name: str = self.__class__.__name__ if name is None else name
+        self.dim: int = dim
+        self.target: Mobject | None = target
+        self.z_index: float = z_index
```

Comment:
It's kind of weird that mypy doesn't infer these? It should be able to :thinking: